### ISSUE TITLE
Correctness of Hessians and Gradients

### ISSUE THREAD

**Issue by keyurja** [microsoft/LightGBM#2423]
From my understanding, for binary objective without sample weight and class weight, we should have (like in xgboost) : gradient = (p - y), hessian = p * (1 - p)

But, in light-gbm we have : gradient = x, hessian = abs(x) * (1 - abs(x)), where x = -y/(1+e^(y*score))

Can someone please help me figure out how these two are similar or work similarly?

**Comment by guolinke** [comment]
y is 1 or -1, so x equals to p.
